src/inputs/internal/SelectFieldBase.tsx

```diff
@@ -283,7 +283,22 @@ export function SelectFieldBase<O, V extends Value>(props: BeamSelectFieldBasePr
     // Only update the fieldset when options change, when options is an array.
     // Otherwise, if the options are passed in as an object, then we assume the caller is updating options via a Promise and not via updating props.
     if (Array.isArray(maybeOptions) && maybeOptions !== fieldState.allOptions) {
-      setFieldState((prevState) => ({ ...prevState, filteredOptions: maybeOptions, allOptions: maybeOptions }));
+      setFieldState((prevState) => {
+        const selectedOptions = maybeOptions.filter((o) => values?.includes(getOptionValue(o)));
+        return {
+          ...prevState,
+          selectedKeys: selectedOptions?.map((o) => valueToKey(getOptionValue(o))) ?? [],
+          inputValue:
+            selectedOptions.length === 1
+              ? getOptionLabel(selectedOptions[0])
+              : multiselect && selectedOptions.length === 0
+              ? nothingSelectedText
+              : "",
+          selectedOptions: selectedOptions,
```

Comment:
Missing these three fieldState properties caused the field not to reflect the correct value.

Reply:
Kinda wonder if we could have like an `updateFieldState(prevState, newValues)` function where `newValues` is a Partial or something, and then any change to `setFieldState`` actually goes through `setFieldState(updateFieldState(prevState, newValues))` so that we ensure all ~3-5 keys are more "reactive" / updated as a group.

Reply:
Yea, I think its a good idea as there is some copy and pasted logic. I think we when decouple our single and multi select fields from `SelectFieldBase`, that'd be a good time to create that helper function. Hopefully that is coming soon.